src/main.rs

```diff
@@ -112,46 +118,71 @@ fn main() {
         warning!(x, opts.greppable, opts.accessible);
     }
 
-    let mut nmap_bench = NamedTimer::start("Nmap");
+    let mut script_bench = NamedTimer::start("Scripts");
     for (ip, ports) in ports_per_ip.iter_mut() {
-        let nmap_str_ports: Vec<String> = ports.into_iter().map(|port| port.to_string()).collect();
+        let vec_str_ports: Vec<String> = ports.into_iter().map(|port| port.to_string()).collect();
 
         // nmap port style is 80,443. Comma separated with no spaces.
-        let ports_str = nmap_str_ports.join(",");
+        let ports_str = vec_str_ports.join(",");
 
-        // if greppable mode is on nmap should not be spawned
-        if opts.greppable || opts.no_nmap {
+        // if option scripts is none, no script will be spawned
+        if opts.greppable || opts.scripts.clone() == ScriptsRequired::None {
             println!("{} -> [{}]", &ip, ports_str);
             continue;
         }
-        detail!("Starting Nmap", opts.greppable, opts.accessible);
-
-        let addr = ip.to_string();
-        let user_nmap_args =
-            shell_words::split(&opts.command.join(" ")).expect("failed to parse nmap arguments");
-        let nmap_args = build_nmap_arguments(&addr, &ports_str, &user_nmap_args, ip.is_ipv6());
-
-        output!(
-            format!(
-                "The Nmap command to be run is nmap {}\n",
-                &nmap_args.join(" ")
-            ),
-            opts.greppable.clone(),
-            opts.accessible.clone()
-        );
+        detail!("Starting Script(s)", opts.greppable, opts.accessible);
+
+        // Run all the scripts we found and parsed based on the script config file tags field.
+        for mut script_f in scripts_to_run.clone() {
+            output!(
+                format!("Script to be run {:?}\n", script_f.call_format,),
+                opts.greppable.clone(),
+                opts.accessible.clone()
+            );
 
-        // Runs the nmap command and spawns it as a process.
-        let mut child = Command::new("nmap")
-            .args(&nmap_args)
-            .spawn()
-            .expect("failed to execute nmap process");
+            // This part allows us to add commandline arguments to the Script call_format, appending them to the end of the command.
+            if opts.command.len() > 0 {
```

Comment:
I think it makes sense, in case the user want to simply append an extra argument to the execution of the scripts. For example in case of an nmap script, we give the possibility to just append extra arguments to RustScan that gets passed to the script. An extra `-sC` switch for nmap will be just a quick append to the RustScan commandline arguments. If we remove it, the users will have to change the script file every time they want to append a new argument to it. This will also improve development time and experience for writing RustScan script files, as the writer of the script can test arguments in terminal and append them to the script execution format if they fit.